simpleviewinc/keg-hub
# Keg-Hub Tasks

## Context
* With the migration to a mono-repo, we need a way to manage the sub repos

## Goal
* Add tasks to the keg-cli to manage the sub repos

## Updates
* Started adding hub tasks
* Added `hub repos` task to print hub repos
* Added `hub run` task to run `package.json script` for sub repos

## Test

**Setup**
* **IMPORTANT** - DO THIS FIRST
  * Pull the branch => `cd ~/keg-hub && keg pr 2`
* Ensure your environment is pointing to the correct keg-cli
  * Open your `~/.bash_profile` and look for this line => `source /Users/<your user>/keg/keg-cli/keg`
  * Edit the line to be this => `source /Users/<your user>/keg-hub/repos/keg-cli/keg`
  * Save and exit the file
* The reload your `.bash_profile` by entering `source ~/.bash_profile`

**Test 1**
* Run the `repos` task
  * `keg hub repos`
  * Ensure it prints out all sub repos in the repos dir
* Run the `repos` task with the options
  * Use the context option to filter which repos are returned
  * `keg hub repos --context core`
    * Only the `keg-core` repo should be printed out
  * `keg hub repos --context co`
    * Both `keg-core` and `keg-components` repos should be printed out

**Test 2**
* Run the `run` task
  * `keg hub run install`
  * Watch in *ahhh* as all `node_modules` are installed for all sub repos
* Run the `run` task with options
  * `keg hub run test cli`
  * Ensure all tests pass
* Run tests for all sub repos
  * `keg hub run test`
  * Ensure all tests pass
  * Right now there's a lot printed out, and it's hard to see what's going on
  * In the future this can be cleaned up so
    * It only prints out failed tests
    * And for CI environments, it will fail and exit the process, but this is a good start for now
